kml/serdes_gpx.cpp

```diff
@@ -131,42 +134,44 @@ std::string const & GpxParser::GetTagFromEnd(size_t n) const
   return m_tags[m_tags.size() - n - 1];
 }
 
-void GpxParser::ParseColor(std::string const & value)
+std::optional<uint32_t> GpxParser::ParseColorFromHexString(std::string_view colorStr)
 {
-  auto const colorBytes = FromHex(value);
-  if (colorBytes.size() != 3)
+  if (colorStr.empty())
   {
-    LOG(LWARNING, ("Invalid color value", value));
-    return;
+    LOG(LWARNING, ("Invalid color value", colorStr));
+    return {};
   }
-  m_color = kml::ToRGBA(colorBytes[0], colorBytes[1], colorBytes[2], (char)255);
-}
-
-// https://osmand.net/docs/technical/osmand-file-formats/osmand-gpx/. Supported colors: #AARRGGBB/#RRGGBB/AARRGGBB/RRGGBB
-void GpxParser::ParseOsmandColor(std::string const & value)
-{
-  if (value.empty())
+  if (colorStr.front() == '#')
+    colorStr.remove_prefix(1);
+  if (colorStr.size() != 6 && colorStr.size() != 8)
   {
-    LOG(LWARNING, ("Empty color value"));
-    return;
+    LOG(LWARNING, ("Invalid color value", colorStr));
+    return {};
   }
-  std::string_view colorStr = value;
-  if (colorStr.at(0) == '#')
-    colorStr = colorStr.substr(1, colorStr.size() - 1);
   auto const colorBytes = FromHex(colorStr);
-  uint32_t color;
   switch (colorBytes.size())
   {
-    case 3:
-      color = kml::ToRGBA(colorBytes[0], colorBytes[1], colorBytes[2], (char)255);
-      break;
-    case 4:
-      color = kml::ToRGBA(colorBytes[1], colorBytes[2], colorBytes[3], colorBytes[0]);
-      break;
-    default:
-      LOG(LWARNING, ("Invalid color value", value));
-      return;
+  case 3: return kml::ToRGBA(colorBytes[0], colorBytes[1], colorBytes[2], (char)255);
+  case 4: return kml::ToRGBA(colorBytes[1], colorBytes[2], colorBytes[3], colorBytes[0]);
+  default:
+    LOG(LWARNING, ("Invalid color value", colorStr));
+    return {};
   }
+}
+
+void GpxParser::ParseColor(std::string_view colorStr)
+{
+  if (const auto parsed = ParseColorFromHexString(colorStr); parsed.has_value())
+    m_color = parsed.value();
+}
+
+// https://osmand.net/docs/technical/osmand-file-formats/osmand-gpx/. Supported colors: #AARRGGBB/#RRGGBB/AARRGGBB/RRGGBB
+void GpxParser::ParseOsmandColor(std::string const & value)
+{
+  std::optional<uint32_t> parsed = ParseColorFromHexString(value);
+  if (!parsed.has_value())
```

Comment:
nit:
```suggestion
  auto const color = ParseColorFromHexString(value);
  if (!color)
```

Should a warning be printed if color fails to parse?

Reply:
There is some logging in GpxParser::ParseColorFromHexString for unexpected values, do you want to add some more logging?